### PR TITLE
Clesaec/tdi 47640 common debt

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TDI-47640 : technical debt of connectors-se/common ; [see sonar analysis](https://sonar-eks.datapwn.com/dashboard?branch=clesaec%2FTDI-47640_CommonDebt&id=org.talend.components%3Aconnectors-se)

### CODE REVIEW

**Review by ypiel-talend — COMMENTED** [review]
Is it possible to detail a scope in the jira ticket ? Lots of things in it. 
Maybe warning messages before and after the changes ?

**Comment by clesaec** [comment]
JIRA updated [with more details](https://jira.talendforge.org/browse/TDI-47640)

File: common/src/main/java/org/talend/components/common/service/http/digest/Nonce.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import lombok.Getter;
+
+public class Nonce {
+
+    private final SecureRandom rnd = new SecureRandom();
+
+    private String lastNonce;
+
+    private long nounceCount;
```
Comment (dmytro-sylaiev) [inline common/src/main/java/org/talend/components/common/service/http/digest/Nonce.java:25]: `nonceCount` or `nounceCount`?

**Review by dmytro-sylaiev — COMMENTED** [review]
LGTM, found several microbes

File: common-stream-io/stream-avro/src/test/java/org/talend/components/common/stream/output/avro/AvroRecordWriterTest.java [hunk 2]
```diff
@@ -75,7 +75,8 @@ void addHeadLess() throws IOException {
                 .requiredInt("ID") //
                 .requiredString("content") //
                 .endRecord();
-        cfg.setAvroSchema(avroSchema.toString(true));
+        final String avrSchema = avroSchema.toString(true);
+        cfg.setAvroSchema(avrSchema);
```
[the remark this answers is not in the document]
Comment (clesaec) [reply common-stream-io/stream-avro/src/test/java/org/talend/components/common/stream/output/avro/AvroRecordWriterTest.java:79]: Help to see elements on step by step debugging on this test code.

File: common/src/main/java/org/talend/components/common/service/http/RedirectService.java [hunk 3]
```diff
@@ -48,7 +46,7 @@ public RedirectContext call(final RedirectContext context) {
         String location = Optional
                 .ofNullable(headers)
                 .map(m -> m.get(LOCATION_HEADER))
-                .filter(l -> !l.isEmpty())
+                .filter((List<String> l) -> l != null && !l.isEmpty())
```
[the remark this answers is not in the document]
Comment (clesaec) [reply common/src/main/java/org/talend/components/common/service/http/RedirectService.java:49]: Indeed, it's never mandatory, but i prefer have explicit type for reading

File: common/src/main/java/org/talend/components/common/service/http/digest/Nonce.java (hunk 1 above)
Comment (clesaec) [reply common/src/main/java/org/talend/components/common/service/http/digest/Nonce.java]: changed

**Review by dmytro-sylaiev — APPROVED** [review]

**Pull request merged**